pkg/util/cmputil/reporter.go

```diff
@@ -0,0 +1,97 @@
+package cmputil
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+// DiffReport is a simple custom reporter that only records differences
```

Comment:
Would it be worth saying that this is specifically an implementation of a `go-cmp` Reporter interface, or linking to https://pkg.go.dev/github.com/google/go-cmp/cmp#Reporter in the doc-comment? It wasn't immediately obvious to me what exactly this was implementing.